yas3fs/__init__.py

```diff
@@ -2624,9 +2635,14 @@ def write(self, path, new_data, offset, fh=None):
         if not self.cache.has(path) or self.cache.is_empty(path):
             logger.debug("write '%s' '%i' '%i' '%s' ENOENT" % (path, len(new_data), offset, fh))
             raise FuseOSError(errno.ENOENT)
-        if isinstance(new_data, unicode): # Fix for unicode
-            logger.debug("write '%s' '%i' '%i' '%s' unicode fix" % (path, len(new_data), offset, fh))
-            new_data = str(new_data.encode('utf-8'))
+
+        if sys.version_info < (3, ):
+            if isinstance(new_data, unicode): # Fix for unicode
```

Comment:
This is puzzling. I'm not sure what the original was intended to do -- the `str()` bit in `str(new_data.encode('utf-8'))` is redundant on Python 2.7. Seems like `new_data` is just supposed to be a bytes object, so can't we just do `new_data.encode('utf-8')` instead?

Reply:
I think it will still need the `isinstance(new_data, bytes)` but, yes, I will remove the `if sys.version_info < (3, )` bit